src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -61,7 +66,8 @@ public BufferReader(in ReadOnlySequence<T> buffer)
         /// <summary>
         /// The current position in the <see cref="Sequence"/>.
         /// </summary>
-        public SequencePosition Position => Sequence.GetPosition(CurrentSpanIndex, _currentPosition);
+        public SequencePosition Position
```

Comment:
@pakrym, is it OK for us to rely on internal implementation details of SequencePosition to construct one (as part of this optimization)?

@JeremyKuhne, will this work regardless of how the ReadOnlySequence was constructed?

Reply:
I don't plan to leave it this way. I'm going to create an issue- chatting with @pakrym I'm going to suggest that we add a Dangerous* method to do this.

Reply:
They have to be part of API review.